pemFioi/quickAlgo/interface-mobileFirst.js

```diff
@@ -249,29 +250,123 @@ var quickAlgoInterface = {
+            "    <div class=\"panel-heading\">" +
+            "        <h2 class=\"sectionTitle\">" +
+            "            <span class=\"iconTag\"><i class=\"icon fas fa-pencil-alt\"></i></span>" +
+                         this.strings.editWindowTitle +
+            "        </h2>" +
+            "    <div class=\"exit\" id=\"editclose\"><i class=\"icon fas fa-times\"></i></div>" +
+            "    </div>" +
+            "    <div class=\"panel-body\">" +
+            "        <div id=\"editExerciseTitle\">" +
+            "            <label>" + this.strings.titleEdition + "</label><input id=\"editExerciseTitleInput\" type=\"text\" value=\"" + document.title + "\"/>" +
+            "        </div>" +
+            "        <div id=\"editExerciseDescription\">" +
+            "            <label>" + this.strings.descriptionEdition + "</label>" +
+            "            <textarea rows=\"10\" id=\"editExerciseDescriptionTextarea\">" + description + "</textarea>" +
+            "        </div>" +
+            "        <div id='panel-body-bottom'>" +
+            "            <button id='saveExerciseChanges'>" + this.strings.saveAndQuit + "</button>" +
+            "        </div>" +
+            "    </div>" +
+            "</div>";
+
+        window.displayHelper.showPopupDialog(editExerciseHtml);
+
+        var that = this;
+
+        $("#editclose").click(function() {
+            var newTitle = $("#editExerciseTitleInput").val();
+            var newDesc = $("#editExerciseDescriptionTextarea").val();
+            var oldTitle = document.title;
+            var oldDescription = $(".exerciseText").first().text();
+            if (newTitle !== oldTitle || newDesc !== oldDescription) {
+                if (!window.confirm(that.strings.quitWithoutSavingConfirmation)) {
+                    return;
+                }
+            }
+
+            $('#popupMessage').hide();
+            window.displayHelper.popupMessageShown = false;
+        });
+
+        $("#saveExerciseChanges").click(function() {
+            $('#popupMessage').hide();
+            window.displayHelper.popupMessageShown = false;
+
+            var newTitle = $("#editExerciseTitleInput").val();
+            var newDesc = $("#editExerciseDescriptionTextarea").val();
+
+            document.title = newTitle;
+            $(".exerciseText").text(newDesc);
+        });
+    },
+
     loadPrograms: function(formElement) {
         this.blocklyHelper.handleFiles(formElement.files);
         resetFormElement($(formElement));
         this.closeEditorMenu();
     },
 
+    /**
+     * This function allow us to save the subject into the additional data saved inside of the interface
+     * This also save the element from the context
+     * @param additional The additional data where we should save subject
+     */
+    saveSubject: function(additional) {
+        if (this.options.canEditSubject) {
+            additional.subject = {
```

Comment:
I would use another name than `subject`, both because that feature might be extended in the future, and because it's odd to have a `subject` object containing a `subject` (and can lead to coding mistakes). `userTaskData` maybe?

Reply:
Have change the variable to userTaskData